src/IniFile.cpp

```diff
@@ -33,81 +33,73 @@ bool IniFile::ReadFile( const RString &sPath )
 	return ReadFile( f );
 }
 
-bool IniFile::ReadFile( RageFileBasic &f )
+bool IniFile::ReadFile(RageFileBasic& f)
 {
 	RString keyname;
-	// keychild is used to cache the node that values are being added to. -Kyz
-	XNode* keychild= nullptr;
-	for(;;)
+	XNode* keychild = nullptr;
+
+	while (true)
 	{
 		RString line;
-		// Read lines until we reach a line that doesn't end in a backslash
-		for(;;)
+		while (true)
 		{
 			RString s;
-			switch( f.GetLine(s) )
+			int result = f.GetLine(s);
+			if (result == -1)
 			{
-			case -1:
 				m_sError = f.GetError();
 				return false;
-			case 0:
-				return true; // eof
+			}
+			else if (result == 0)
+			{
+				return true;
 			}
 
-			utf8_remove_bom( s );
+			utf8_remove_bom(s);
 
 			line += s;
 
-			if( line.empty() || line[line.size()-1] != '\\' )
+			if (line.empty() || line.back() != '\\')
 			{
 				break;
 			}
-			line.erase( line.end()-1 );
+			line.pop_back();
 		}
 
+		if (line.empty())
+			continue;
 
-		if( line.empty() )
+		if (line[0] == ';' || line[0] == '#' || (line.size() > 1 && line[0] == line[1] && (line[0] == '/' || line[0] == '-')))
```

Comment:
Personally I think these changes seem harder to read than the swtich. I think I would prefer to keep the switch style as in the original file/

Reply:
That is fine with me.